lineman/app/services/ability_service.coffee

```diff
@@ -68,3 +68,19 @@ angular.module('loomioApp').factory 'AbilityService', (CurrentUser) ->
 
     canManageMembershipRequests: (group) ->
       (group.membersCanAddMembers and CurrentUser.isMemberOf(group)) or @canAdministerGroup(group)
+
+    canViewGroup: (group) ->
+      group.visibleToPublic() or
+      CurrentUser.isMemberOf(group) or
+      (group.visibleToOrganisation() and CurrentUser.isMemberOf(group.parent()))
+
+    canJoinGroup: (group) ->
+      (group.membershipGrantedUpon == 'request') and
+      @canViewGroup(group) and
+      !CurrentUser.isMemberOf(group)
+
+    canRequestMembership: (group) ->
```

Comment:
Yarr, abilities are a good thing to be testing.
I wonder how many times we'd need to be including multiple ability dependencies within a controller if we broke it up? I'm thinking something like

```
.controller($scope, GroupAbility, DiscussionAbility, MembershipAbility) ->
```

would make me sad.